content/influxdb/v2.0/query-data/common-queries/compare_last_measurement.md

```diff
@@ -0,0 +1,46 @@
+---
+title: Compare the last measurement to another bucket's mean
+seotitle: Compare the last measurement to a mean stored in another bucket
+description: >
+  Compare the value from the latest point to an average value stored in another bucket. This is useful when using the average value to calculate a threshold check.
+influxdb/v2.0/tags: [queries]
+menu:
+  influxdb_2_0:
+    name: Compare values from different buckets
+    parent: Common queries
+weight: 104
+---
+
+{{% note %}}
+This example uses [NOAA water sample data](/influxdb/v2.0/reference/sample-data/#noaa-water-sample-data).
+{{% /note %}}
+
+Compare the value from the latest point to an average value stored in another bucket. This is useful when using the average value to calculate a [threshold check](/influxdb/v2.0/monitor-alert/checks/create/#threshold-check).
+
+The following query:
+  - Gets the last value in the `means` bucket and compares it to the last value in the `noaa` bucket using [`last()`](/influxdb/v2.0/reference/flux/stdlib/built-in/transformations/selectors/last/).
+  - Uses [`join()`](/influxdb/v2.0/reference/flux/stdlib/built-in/transformations/join/) to combine the results
+  - Uses [`map()`](/influxdb/v2.0/reference/flux/stdlib/built-in/transformations/map/) to calculate the differences
+
+  ```js
+  means = from(bucket: "weekly_means")
+    |> range(start: 2019-09-01T00:00:00Z)
+    |> last()
+    |> keep(columns: ["_value", "location"])
+
+  latest = from(bucket: "noaa")
+    |> range(start: 2019-09-01T00:00:00Z)
+    |> filter(fn: (r) => r._measurement == "average_temperature")
+    |> last()
+    |> keep(columns: ["_value", "location"])
+
+  join(tables: {mean: means, reading: latest}, on: ["location"])
+    |> map(fn: (r) => ({r with deviation: r._value_reading - r._value_mean}))
+  ```
```

Comment:
- This code block doesn't need to be indented.
- Sidenote – Markdown is really bad at knowing when a block element like a code block should be nested in a list. The way this is formatted, this code block will be rendered as part of the list above. Add an empty html comment `<!-- -->` just before the code block to force it to render correctly:

~~~md
<!-- -->
```js
...
~~~